File: demo/src/main/java/com/midas/demo/mvp/model/RankModel.java
package com.midas.demo.mvp.model;

import com.midas.demo.mvp.model.bean.HomeBean;
import com.midas.demo.net.RetrofitManger;
import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;


public class RankModel {

    public Observable<HomeBean.Issue> getRankList(String url){
        return RetrofitManger.getApiService()
                .getIssueData(url)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread());
    }
}

File: app/src/main/java/com/midas/eye/mvp/model/RankModel.kt
package com.midas.eye.mvp.model

import com.midas.eye.mvp.model.bean.HomeBean
import com.midas.eye.net.RetrofitManager
import com.midas.eye.rx.scheduler.SchedulerUtils
import io.reactivex.Observable

/**
 * Created by midas on 2019/05/28.
 * desc: 排行榜 Model
 */
class RankModel {

    /**
     * 获取排行榜
     */
    fun requestRankList(apiUrl:String): Observable<HomeBean.Issue> {
        return RetrofitManager.service.getIssueData(apiUrl)
                .compose(SchedulerUtils.ioToMain())
    }

}
